File: Trinity.Runtime/platform_src/ios/elastOS/Classes/ContactNotifier/db/ReceivedInvitation.swift
import SQLite

public class ReceivedInvitation {
    public var iid: Int // Invitation unique ID
    public var did: String
    public var carrierUserID: String
    
    public static let invitationIdField = Expression<Int64>(CNDatabaseHelper.INVITATION_ID)
    public static let didField = Expression<String>(CNDatabaseHelper.DID)
    public static let carrierUserIdField = Expression<String>(CNDatabaseHelper.CARRIER_USER_ID)
    
    init() {
        iid = -1
        did = ""
        carrierUserID = ""
    }

    /**
     * Creates a ReceivedInvitation object from a RECEIVED_INVITATIONS row.
     */
    static func fromDatabaseCursor(row: Row) -> ReceivedInvitation {
        let invitation = ReceivedInvitation()
        invitation.iid = Int(row[invitationIdField])
        invitation.did = row[didField]
        invitation.carrierUserID = row[carrierUserIdField]
        return invitation
    }
}

File: Trinity.Runtime/platform_src/android/app/src/main/java/org/elastos/trinity/runtime/contactnotifier/db/ReceivedInvitation.java
package org.elastos.trinity.runtime.contactnotifier.db;

import android.database.Cursor;

public class ReceivedInvitation {
    public int iid; // Invitation unique ID
    public String did;
    public String carrierUserID;

    /**
     * Creates a ReceivedInvitation object from a RECEIVED_INVITATIONS row.
     */
    static ReceivedInvitation fromDatabaseCursor(Cursor cursor) {
        ReceivedInvitation invitation = new ReceivedInvitation();
        invitation.iid = cursor.getInt(cursor.getColumnIndex(DatabaseHelper.INVITATION_ID));
        invitation.did = cursor.getString(cursor.getColumnIndex(DatabaseHelper.DID));
        invitation.carrierUserID = cursor.getString(cursor.getColumnIndex(DatabaseHelper.CARRIER_USER_ID));
        return invitation;
    }
}
